src/AutoMapper/ConfigurationValidator.cs

```diff
@@ -1,15 +1,53 @@
 using System;
 using System.Collections.Generic;
+using System.ComponentModel;
 using System.Linq;
 using AutoMapper.Configuration;
 
 namespace AutoMapper
 {
+    using Validator = Action<ValidationContext>;
+
+    [EditorBrowsable(EditorBrowsableState.Never)]
```

Comment:
We're not Microsoft :) I don't think we should have the same standards. It's more work for us and advanced users will be frustrated. I'm in the same position vs some MS libraries.

Reply:
Should we move these to a different namespace then? That might make things a bit more obvious than using `EditorBrowsable`.

Reply:
The reason why I say this is because I get notes in my email about breaking public APIs after we release minor versions :)

I'd rather be very explicit about what is public or not, instead of breaking people.

Reply:
Changing the namespace is a breaking change and it doesn't work with ApiCompat. There are many such classes. This people are being disingenuous. I'm a competent developer and I would never raise such claims, even with MS. Dig inside some assembly, use smth undocumented and then complain when it breaks. Would _you_ do that? And it spoils things for well intended people that just want to use some internals, knowing very well that any update might break their code. So I'm not against changing the namespace, but we still need `EditorBrowsable` as far as I can tell.

Reply:
The problem these folks run into is not that *they're* using the APIs, but some dependency they have does. So they upgrade from x.0 to x.1, but their dependency has not upgraded to x.1 which has a breaking API change. Check the "Used By" section in GitHub, and the folks using those Used By packages are the ones that complain when we break the dependencies.

Reply:
I'm OK breaking our APIs - just want to limit the exposure of doing so and make it obvious. I'm not sure a lack of API docs are enough to say "this might break"

Reply:
OK, there five API changes. In my view nobody should depend on any of that. Other than, you know, people at the end of their rope :) If you disagree, which ones seem problematic, and let's see if we can change the namespace for them or smth else.

Reply:
Looking more closely, mainly I removed stuff. So the only thing to do might be to keep `NullableDestinationExpressionBinder` as a stub. Is it worth it?